### PR TITLE
Net processing: Only call PushAddress() from net_processing

### PR DESCRIPTION
**Pull request by jnewbery**
This is the first part of #21186. It slightly disentangles addr handling in net/net_processing by making it explicit that net_processing is responsible for pushing addr records into `vAddrToSend`.

### CODE REVIEW

**Comment by jnewbery** [comment]
Rebased. Moving out of draft status.

**Review by sdaftuar — APPROVED** [review]
utACK, just nits.

File: src/net.cpp [hunk 1]
```diff
@@ -222,10 +221,12 @@ void AdvertiseLocal(CNode *pnode)
         }
         if (addrLocal.IsRoutable() || gArgs.GetBoolArg("-addrmantest", false))
         {
-            LogPrint(BCLog::NET, "AdvertiseLocal: advertising address %s\n", addrLocal.ToString());
-            pnode->PushAddress(addrLocal, rng);
+            LogPrint(BCLog::NET, "GetPeerLocalAddr: advertising address %s\n", addrLocal.ToString());
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply src/net.cpp]: Done

File: src/net.cpp [hunk 2]
```diff
@@ -201,8 +201,7 @@ bool IsPeerAddrLocalGood(CNode *pnode)
            IsReachable(addrLocal.GetNetwork());
 }
 
-// pushes our own address to a peer
-void AdvertiseLocal(CNode *pnode)
+Optional<CAddress> GetPeerLocalAddr(CNode *pnode)
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply src/net.cpp]: Yes, that's better. Changed.

**Comment by jnewbery** [comment]
Thanks for the review @sdaftuar! I've taken both of your suggestions.

**Comment by sdaftuar** [comment]
utACK 3c0325e7137f0423c75326b25faa856c5f67b7eb

File: src/net.cpp [hunk 3]
```diff
@@ -222,10 +221,12 @@ void AdvertiseLocal(CNode *pnode)
         }
         if (addrLocal.IsRoutable() || gArgs.GetBoolArg("-addrmantest", false))
         {
-            LogPrint(BCLog::NET, "AdvertiseLocal: advertising address %s\n", addrLocal.ToString());
-            pnode->PushAddress(addrLocal, rng);
+            LogPrint(BCLog::NET, "GetLocalAddrForPeer: advertising address %s to peer=%d\n", addrLocal.ToString(), pnode->GetId());
```
Comment (MarcoFalke) [inline src/net.cpp]: in the first commit: Would be good to remove the function name when touching this. You can use #19809 to get the function name in the log message.

**Review by maflcko — APPROVED** [review]
cr ACK 3c0325e7137f0423c75326b25faa856c5f67b7eb 🔼

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

cr ACK 3c0325e7137f0423c75326b25faa856c5f67b7eb 🔼
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhx2Av/ciCyJfD1yNDR55VEg+5wswy1krjwSeDjeJIlxwwbr5pH5GGB0opCpf7Z
OMODIQ9/qtqNnerNcqLvAuru+waN4pBOFgV6qkoAfCPjsfwfqwz7BHXmfdbOVD3g
how3qn2gsEKoxPPz9E1Ei7yJeI9a2sg5fmpaWVI8rWXtLFRvIkT9PC0jRauJyRY9
EsW+OCKnb0LCcmg+DlVvGx/S9op4vnV2e2LHSPaSoEdSnDZ73HxxDn7Z5mMuwQZV
fceOyU0JAQKvK7HCL/cKffrCnTNX79KD0uRL6mxPQwssM6N4W4CIYaXosj83KBgt
FtU3SxjFg4GcVChqLLjWXlZt2iCw3ApheqWRJwsWLJgRvKUL8pmsLsOipDT40r6j
8dX/BCWWJg+XlJg4vvY7UPHWNZjWW7BW99fDv7ejeUk8tMldyRbX8DQZSCuaSp96
CsAr0KtRsuvxfJquxH7vIdizUK0aYrXS/MFyhsSPswwWxQXbljViWMDYuYH+5DLh
LatvFqnM
=sO8d
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `11e4333572630de023790d12cb37c30526dcd13d95c1e10af4452856fb74e1a7  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e89294010811e4333572630de023790d12cb37c30526dcd13d95c1e10af4452856fb74e1a7f0107c477f2c1d82d284237137250670a15e08fff010285d4ec370b006041713a40235b334ac08f104602e2b16f0089806e1e8f4e580c00083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff0109af6ce2b3ac576563ff23c56f40211da08f1206f023318ba6b2d06fbc9bea50d54b516961085461d84362d851d13011011b0c408f104602e1d43f008373ade5fa07410db0083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff010ac4b9e500572a3d6560418cbbae0c4bb08f120ba32c2546012d813ed8c31dd365f1c391a987838a980ae355374da04423f08aa08f104602e2b15f008902c6fda6a3e23030083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010ccce531eeff3e9749e7a313c4b1473ae08f104602e2b16f008b61bc37f1558188a0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

File: src/net.cpp (hunk 3 above)
Comment (jnewbery) [reply src/net.cpp]: Done

**Comment by jnewbery** [comment]
Thanks for the review @MarcoFalke! I've taken your suggestion.

**Review by maflcko — APPROVED** [review]
re-ACK 3e68efa615968e0c9d68a7f197c7852478f6be78 🍅

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

re-ACK 3e68efa615968e0c9d68a7f197c7852478f6be78 🍅
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUiTQAv/UHXsGIiH4mmC9FCK7uV0sMXzxWfaNPXlbzc8HRJ5Q5BxO/hpC+vQSpNN
Rcg4eWWI+GdpqeX2EXyLCUTckSfQGOtCc+TRztjjYzWIcXojpV/XstKDgqPXj4kk
hIAa4BH67gYN3XSX5qqCVwtqcxiCAfb7lGi3oQ2wH/qaQXylIe9lPTx8Ijm1iXtD
iks+MHRcUhzhKbWIGcY+3lxIAZn7QiBLEyR8FQUYqf4tG1oNftMb8lwaml/ti1GC
IlxB1ast/LIkSNW4boMHX+QnxODoK+hLteY+7qdUQyZ732wlhq75vWIjNoEVRpki
G7P/1lPZvsQ47iprz0TBtnrSvZbK80uKtpX+IQsVyMku6brNyA7eB7rbeIir9awu
TXFMt19dR/VwfX+5BadfcmIAso9S4tAGoKLBb86BosaN21TUxqwt+sxcLyFO5bv8
vw9tItFxMlpbdJ1fOl39qZ/eiaITn6QyJEfoS7S7TNrdrcENbfJQJdW8xfymbnwQ
cd7wi0jV
=z/AJ
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `b508a08160a657b0e132e8bbdb91d64571be968c585f6ca2a374baecd1fb4875  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108b508a08160a657b0e132e8bbdb91d64571be968c585f6ca2a374baecd1fb4875f0101d08efe0d96e0087620cea07ddf03d3d08fff0103e76213219ec41938dd33f5740866f2608f1202504da9384c99e1305f6e0681690150da6dc583adb34c465dcd2984c5a22357908f104602e76d4f0081209e31ae3ddcde30083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff01048e96a22f68dd34765a4f1ab6943380708f02089412cd64737e5dafa0cbf3ec0be1254b0ce57be95e65272d95bd6544a2bb6e208f104602e6902f0084e892e23e761bd610083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff0104a612205d5c8680e1a3232cded49470708f104602e76d5f008f5981810e688c6750083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010b68364905d3d1ddeca9d8f81b751be9708f104602e76d5f008328bc58b83093e000083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Comment by sdaftuar** [comment]
re-ACK

**Pull request merged**